cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaCluster.java

```diff
@@ -478,6 +489,22 @@ public static List<String> generatePodList(String cluster, int replicas) {
         return podNames;
     }
 
+    /**
+     * If the cluster already exists and has a cluster ID set in its status, it will be extracted from it. If it doesn't
+     * exist in the status yet, a new cluster ID will be generated. The cluster ID is used to bootstrap KRaft clusters.
+     *
+     * @param kafkaCr   The Kafka CR from which the cluster ID should be extracted
+     *
+     * @return  The extracted or generated cluster ID
+     */
+    private static String getOrGenerateKRaftClusterId(Kafka kafkaCr)   {
+        if (kafkaCr.getStatus() != null && kafkaCr.getStatus().getClusterId() != null) {
+            return kafkaCr.getStatus().getClusterId();
+        } else {
+            return Uuid.randomUuid().toString();
```

Comment:
Pardon my ignorance, but where does the cluster ID get added to the status?

Reply:
It is done here: https://github.com/strimzi/strimzi-kafka-operator/blob/9850f22e169fd19572ecefe7d5bde393e452fefd/cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaReconciler.java#L1126-L1158

It is not part of this PR. We have this already from the past, but it used to be more for informative purposes for the user. Not it is used to also carry the cluster ID for the operator. I will add it to the Javadoc.